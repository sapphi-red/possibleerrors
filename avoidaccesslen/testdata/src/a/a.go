package a

import "fmt"

func f() {
	slice := []string{}
	fmt.Println(slice[len(slice)]) // want "Will occur index out of range"

	slice2 := slice
	fmt.Println(slice2[len(slice2)]) // want "Will occur index out of range"

	fmt.Println(slice[len(slice)-1])
	fmt.Println(slice[len(slice)+1]) // want "Will occur index out of range"
	fmt.Println(slice[len(slice)+2]) // want "Will occur index out of range"

	fmt.Println(slice[-1+len(slice)])
	fmt.Println(slice[1+len(slice)])   // want "Will occur index out of range"
	fmt.Println(slice[3-2+len(slice)]) // want "Will occur index out of range"
	fmt.Println(slice[3+4+len(slice)]) // want "Will occur index out of range"

	sliceA := []string{}
	fmt.Println(sliceA[len(slice)])
	fmt.Println(sliceA[len(slice)+1])
}
