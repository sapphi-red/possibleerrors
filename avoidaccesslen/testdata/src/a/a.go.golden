package a

import "fmt"

func f() {
	slice := []string{}
	fmt.Println(slice[len(slice)-1]) // want "Will occur index out of range"

	slice2 := slice
	fmt.Println(slice2[len(slice2)-1]) // want "Will occur index out of range"

	slice3 := []string{}
	fmt.Println(slice3[len(slice)])
}
